packages/at_secondary_server/lib/src/verb/handler/pkam_verb_handler.dart

```diff
@@ -86,25 +87,31 @@ class PkamVerbHandler extends AbstractVerbHandler {
       String enrollId, String atSign) async {
     String enrollmentKey =
         '$enrollId.$newEnrollmentKeyPattern.$enrollManageNamespace$atSign';
-    var enrollData = await keyStore.get(enrollmentKey);
-    final atData = enrollData.data;
-    final enrollDataStoreValue =
-        EnrollDataStoreValue.fromJson(jsonDecode(atData));
-    EnrollStatus enrollStatus =
-        EnrollStatus.values.byName(enrollDataStoreValue.approval!.state);
-
+    late final EnrollDataStoreValue enrollDataStoreValue;
     ApkamVerificationResult apkamResult = ApkamVerificationResult();
-    apkamResult.response = _getApprovalStatus(
-        enrollStatus, enrollId, enrollDataStoreValue.approval!.state);
+    AtData? enrollData;
+    try {
+      enrollData = await keyStore.get(enrollmentKey);
+      enrollDataStoreValue =
+          EnrollDataStoreValue.fromJson(jsonDecode(enrollData!.data!));
+      EnrollStatus enrollStatus =
+          EnrollStatus.values.byName(enrollDataStoreValue.approval!.state);
+      apkamResult.response = _getApprovalStatus(enrollStatus, enrollId);
+    } on KeyNotFoundException catch (e) {
+      logger.finer('Caught: $e');
+      apkamResult.response = _getApprovalStatus(EnrollStatus.expired, enrollId);
```

Comment:
if there is an exception, rest of the code below should not execute

Reply:
changed in [04c08a2](https://github.com/atsign-foundation/at_server/pull/1536/commits/04c08a2a6c3c5185aaccc7f02f1cdac7ea55d1eb)